packages/lesswrong/server/resolvers/userResolvers.ts

```diff
@@ -535,3 +535,24 @@ defineQuery({
     return accessFilterMultiple(currentUser, Users, matchedUsers, context);
   }
 });
+
+defineQuery({
+  name: "GetDialogueRecommendedUsers",
+  resultType: "[User]!",
+  fn: async (root, _, context) => {
+    const { currentUser } = context
+    if (!currentUser) {
+      throw new Error('User must be logged in to get recommended users');
+    }
+
+    const upvotedUsers = await context.repos.users.getUsersTopUpvotedUsers(currentUser, 50, 720)
+    let recommendedUsers = await context.repos.users.getDialogueRecommendedUsers(currentUser._id, upvotedUsers);
```

Comment:
`recommendedUsers` isn't getting reassigned, so it doesn't need to be `let` rather than `const`

Reply:
(well, I just saw it get reassigned down below... but you don't need to, you can just slice `shuffled`)

Reply:
Actually I think you can just `ORDER BY random()` in a postgresql query so you probably don't need most of this logic at all.

Reply:
That's cool to know! I actually find it marginally more legible and compatible with future ideas I migth have, to keep the randomisation inside of userResolvers, so will keep it there for now

Reply:
@lagerros This kind of thing can be quite bad for performance so I would strongly recommend against unless you have an immediate use-case which requires pulling 100 users from the DB